Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm

```diff
@@ -8457,6 +8457,32 @@ - (void)doAfterEditorStateUpdateAfterFocusingElement:(dispatch_block_t)block
     _actionsToPerformAfterEditorStateUpdate.append(makeBlockPtr(block));
 }
 
+- (void)_reconcileEnclosingScrollViewContentOffset:(WebKit::EditorState&)state
+{
+    if (!state.hasVisualData())
+        return;
+
+    auto scrollingNodeID = state.visualData->enclosingScrollingNodeID;
+    if (!scrollingNodeID)
+        return;
+
+    RetainPtr scroller = [self _scrollViewForScrollingNodeID:*scrollingNodeID];
+    if (!scroller)
+        return;
+
+    auto possiblyStaleScrollPosition = state.visualData->enclosingScrollPosition;
+    auto scrollDelta = possiblyStaleScrollPosition - WebCore::roundedIntPoint([scroller contentOffset]);
+
+    static constexpr auto adjustmentThreshold = 1;
+    if (std::abs(scrollDelta.width()) <= adjustmentThreshold && std::abs(scrollDelta.height()) <= adjustmentThreshold)
+        return;
+
+    CGSize contentDelta = [scroller convertRect:CGRectMake(0, 0, scrollDelta.width(), scrollDelta.height()) toView:self].size;
+    contentDelta.width *= scrollDelta.width() >= 0 ? 1 : -1;
+    contentDelta.height *= scrollDelta.height() >= 0 ? 1 : -1;
```

Comment:
I think we can use `std::copysign` here?

Reply:
Yep, perfect use case for that!